packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -503,21 +494,34 @@ def _init_chat_handlers(self, ychat: YChat) -> Dict[str, BaseChatHandler]:
                 )
                 continue
 
+            # Skip disabled entrypoints
+            ep_disabled = getattr(chat_handler, "disabled", False)
+            if ep_disabled:
+                self.log.warn(
+                    f"Skipping registration of chat handler `{chat_handler_ep.name}` as it is explicitly disabled."
+                )
+                continue
+
```

Comment:
All of the validation checks should also be moved to a new function in a new module, e.g. `verify_chat_handler_class()` in `jupyter_ai/chat_handlers/utils.py`.

Reply:
Can be done later.